presto-native-execution/presto_cpp/presto_protocol/Base64Util.cpp

```diff
@@ -104,6 +104,11 @@ velox::VectorPtr readScalarBlock(
   for (auto i = 0; i < positionCount; i++) {
     if (!rawNulls || !velox::bits::isBitNull(rawNulls, i)) {
       rawBuffer[i] = stream.read<T>();
+      if (type->isHugeint() && rawBuffer[i] < 0) {
```

Comment:
We have to check for long decimal type. Not all hugeints need this conversion.

Reply:
This might work for now since decimal types are the only hugeints. But with UUID type this will change.